### ISSUE TITLE
reposition Niagara's search to the bottom

### ISSUE THREAD

**Issue by yotamguttman** [8bitPit/Niagara-Issues#2681]
### Is your feature request related to a problem?

currently the search bar is situated at the top of the screen
![Screenshot_20240509-100015](https://github.com/8bitPit/Niagara-Issues/assets/92977002/c74da186-fb96-471e-9b29-5f2d405328fa)

I fell in love with Niagara launcher because it's the most thumb friendly app out there! this is the only shortcoming—the travel distance between the keyboard and the searched apps is too big.

### Describe the solution you'd like

I'd suggest you add an option in the settings to reposition the search box and flip the whole search page layout.

the search box should be placed right atop the keyboard. this way the user can easily glance at what they're typing.

right above the search box, the search results would appear. so that the lowest one (nearest to the search box and the typing finger), is the most relevant and so on. 

it's basically a whole reverse. perhaps some users would prefer it one way or the other. that's why I'd add the option to choose. 
generally, I think the layout I'm suggesting is more intuitive and inlined with the Niagara launcher design philosophy, as it'd make the search feature entirely operatable using one finger.

### Feature Request Procedure

- [X] I have read the [knowledge base article on feature requests](https://help.niagaralauncher.app/article/44-feature-requests) and followed the steps mentioned

**Comment by NiagaraSupport** [comment]
Hey there, thanks for the suggestion.
I'll bring it up for discussion. Visual customization requests, however, typically have a low chance of approval. The search function works well as it is right now, and adjusting the screen layout might disrupt even more users who prefer the current setup.

**Comment by yotamguttman** [comment]
Niagara's design philosophy suggests that this project is about innovation and taking design to new heights and stretching it to new limits. I love it that Niagara was so brave to introduce a totally new way for users to interact with their mobile. a way that's far more intuitive, cleaner, and simpler than any other default launcher.

I recorded my screen to show the usage and the movement of my finger, relative to its place, to demonstrate why my suggestion is actually viable to consider:
![image](https://github.com/8bitPit/Niagara-Issues/assets/92977002/4bf721d3-bd6c-41dc-bb9e-da16378e28c7)

maybe if mobile phones hadn't been so tall nowadays, it wouldn't have been an issue. but check out the travel distance between my typing finger, that's at the bottom of the screen, to the relevant search results that appear all the way at the top and are expected to be tapped on. you added the nice 'tap enter to launch' gesture, but this applies just to the most relevant option. if you displayed the search results from bottom to top, all the relevant apps would be just a little thumb stretch away from where it already is typing.

as I mentioned before, I'm also unable to simultaneously view what I'm typing, as I'm looking at the letters that are at the bottom and the actual text box is on the other end of the display. had the search box been right above the keyboard, I'd be able to type while reviewing the output at once.

I drew a little example:
![Board](https://github.com/8bitPit/Niagara-Issues/assets/92977002/c8f0409c-10d1-42c1-93f0-55299bdaa9ee)


as you suggested, some users might prefer it one way or the other and it's generally good to offer support for both. I bet my suggestion isn't so outlandish. many users will find interacting with the search feature single-handedly, quicker, efficient and a whole lot more comfortable :)

**Comment by NiagaraSupport** [comment]
I appreciate your input. Currently, the idea is being considered for future implementation.

**Comment by kylepantall** [comment]
This idea is excellent. From a UI design perspective, actions that can be performed one-handed, like swipe to search, should be designed for one-handed operation. Furthermore, elements meant for one-handed use should be placed within easy reach of the user's hand on a touch screen device to ensure comfortable and efficient interactions

**Comment by yotamguttman** [comment]
@NiagaraSupport I'm glad to see you're so attentive to your users' needs and ideas :) 

@kylepantall you're right :) I think you should at least offer the option, considering it isn't the most conventional solution and some people are slower to pick up changes than others. but in terms of user experience, there's no doubt it's more practical and easier to use. but it's important to remember that my choice of software like Niagara was driven by the fact that it is unconventional and unafraid to explore new design solutions for even mundane things.

**Comment by contact-maximux** [comment]
This idea is far above than normal feature . It will enhance one hand usage. Alas Niagra team could not consider it.

**Comment by swiftwind501** [comment]
Seconding this suggestion. Flip foldables are especially taller than other phones, so adding this would save the trouble of moving my whole hand to the top of the screen to select a result. In fact, make it the default so our thumbs can catch a break.

**Comment by amerikiwi93** [comment]
Agreed. It would be great to be able to use the launcher with only my thumb without having to stretch to reach the top of the screen.

**Comment by cco** [comment]
Please develop this feature. Thank you. Top bar widgets are not as ergonomic. I simply want a bottom search bar.
